builtin/credential/cert/path_login.go

```diff
@@ -85,6 +78,86 @@ func (b *backend) pathLogin(
 	return resp, nil
 }
 
+func (b *backend) pathLoginRenew(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	config, err := b.Config(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	if config == nil || config.VerifyCert {
+		var matched *ParsedCert
+		if verifyResp, resp, err := b.verifyCredentials(req); err != nil {
+			return nil, err
+		} else if resp != nil {
+			return resp, nil
+		} else {
+			matched = verifyResp
+		}
+
+		if matched == nil {
+			return nil, nil
+		}
+
+		clientCerts := req.Connection.ConnState.PeerCertificates
+		if len(clientCerts) == 0 {
+			return logical.ErrorResponse("no client certificate found"), nil
+		}
+		skid := base64.StdEncoding.EncodeToString(clientCerts[0].SubjectKeyId)
+		akid := base64.StdEncoding.EncodeToString(clientCerts[0].AuthorityKeyId)
+
+		// Certificate should not only match a registered certificate policy but it should match the exact same
+		// certificate which was used to login
```

Comment:
This comment is not quite right -- it will match the same keys used for issuing the certificate and the certificate itself. We are essentially verifying identity rather than verifying the literal same cert.